### ISSUE TITLE
[vcpkg_cmake_config_fixup] allow custom TARGET_PATH

### ISSUE THREAD

**Issue by bansan85** [microsoft/vcpkg#40653]
**Describe the bug**

From a vcpkg port, I run a `pip install`. So everything is installed in `${CURRENT_PACKAGES_DIR}/tools/python3/Lib/site-packages`.

I have warning :

```
[cmake] ...\vcpkg-registry\ports/PROJECT/0.6.26.1\portfile.cmake: warning: There should be no absolute paths, such as the following, in an installed package. To suppress this message, add set(VCPKG_POLICY_SKIP_ABSOLUTE_PATHS_CHECK enabled)
[cmake] note: D:\vcpkg\packages
[cmake] note: ...\build\vcpkg_installed
[cmake] note: D:\vcpkg\buildtrees
[cmake] note: D:\vcpkg\downloads
[cmake] note: Absolute paths were found in the following files
[cmake] D:\vcpkg\packages\PROJECT_x64-wsmep: note: the files are relative to ${CURRENT_PACKAGES_DIR} here
[cmake] note: debug/tools/python3/Lib/site-packages/PROJECT/lib/cmake/PROJECT-0.6.26.1_static/PROJECTConfig.cmake
[cmake] note: debug/tools/python3/Lib/site-packages/PROJECT/lib/cmake/PROJECT-0.6.26.1_static/PROJECTTargets-relwithdebinfo.cmake
[cmake] note: tools/python3/Lib/site-packages/PROJECT/lib/cmake/PROJECT-0.6.26.1_static/PROJECTConfig.cmake
[cmake] note: tools/python3/Lib/site-packages/PROJECT/lib/cmake/PROJECT-0.6.26.1_static/PROJECTTargets-relwithdebinfo.cmake
[cmake] ...\vcpkg-registry\ports/PROJECT/0.6.26.1\portfile.cmake: warning: Found 5 post-build check problem(s). These are usually caused by bugs in portfile.cmake or the upstream build system. Please correct these before submitting this port to the curated registry.
```

To fix it, I need to launch

```
  vcpkg_cmake_config_fixup(
    PACKAGE_NAME "PROJECT" CONFIG_PATH
    "tools/python3/Lib/site-packages/PROJECT/lib/cmake" 
  )
```

My problem it that all CMake files are moved to  `${CURRENT_PACKAGES_DIR}/share` instead of `${CURRENT_PACKAGES_DIR}/tools/python3/Lib/site-packages/PROJECT/share`

**Expected behavior**

Be able to use `vcpkg_cmake_config_fixup` so that all CMake files are moved to `${CURRENT_PACKAGES_DIR}/tools/python3/Lib/site-packages/PROJECT/share`.

**Solution**

Patch `vcpkg-cmake-config` to support following syntax:

```
  vcpkg_cmake_config_fixup(
    PACKAGE_NAME "PROJECT" CONFIG_PATH
    "tools/python3/Lib/site-packages/PROJECT/lib/cmake/PROJECT-${VERSION}${suffix}"
    TARGET_PATH "tools/python3/Lib/site-packages/PROJECT/share/PROJECT"
  )
```

Fixing `vcpkg_cmake_config_fixup` is quite easy. If my problem and my solution are fine, I can make a PR.

**Comment by FrankXie05** [comment]
cc @vicroms @BillyONeal Could you please help take a look?

**Comment by BillyONeal** [comment]
>My problem it that all CMake files are moved to ${CURRENT_PACKAGES_DIR}/share instead of ${CURRENT_PACKAGES_DIR}/tools/python3/Lib/site-packages/PROJECT/share

If they were in that location they can't be found, so I'm not sure what you're trying to do with that.

**Comment by bansan85** [comment]
I want to create two Python modules A and B. B depends on A.

A and B are in C++ and uses external dependencies. These dependencies are managed by vcpkg.

When building B module, I would like to use the A module with its header and its CMake target files.

As I use `pip install .` to build the B module, I need to use the A module installed in `${CURRENT_PACKAGES_DIR}/tools/python3/Lib/site-packages/PROJECT_A`.

I will start working on this problem today. I will tell you if adding a `TARGET_PATH` parameter in `vcpkg_cmake_config_fixup` is sufficient or the good solution.

**Comment by BillyONeal** [comment]
The problem is that `vcpkg_cmake_config_fixup`'s purpose is to make the resulting configs work from vcpkg's toolchain file, and it will never ever look in `${CURRENT_PACKAGES_DIR}/tools/python3/Lib/site-packages`. It will only look in `${CURRENT_PACKAGES_DIR}/share`, which is why `vcpkg_cmake_config_fixup` moves the files there. Is pip trying to invoke cmake somehow?

**Comment by bansan85** [comment]
Yes, pip invokes CMake via the extension `cmake-build-extension`.

**Comment by bansan85** [comment]
It's more complicated than I thought. I want to use the system python.

So I don't want to install in `${CURRENT_PACKAGES_DIR}/tools/python3/Lib/site-packages` but in the system python (which is a virtual environment in my case).

I don't think that vcpkg has a way to know which is the current python executable.

**Comment by dg0yt** [comment]
> I don't think that vcpkg has a way to know which is the current python executable.

For ports depending on a particular version of python, a dependency on `python3` is the key to ABI tracking and the injection point for user-provided overlay ports.

Things get tricky when the target triplet is not the host triplet.
